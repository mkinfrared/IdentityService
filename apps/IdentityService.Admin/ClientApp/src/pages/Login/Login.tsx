import { Button } from "@identity-service/ui";
import { memo, useCallback } from "react";

import userManager from "utils/userManager";

import css from "./Login.module.scss";

const Login = () => {
  const handleClick = useCallback(() => {
    userManager.signinRedirect();
  }, []);

  return (
    <div className={css.Login} data-testid="Login">
      <Button onClick={handleClick}>Login</Button>
    </div>
  );
};

export { Login };

export default memo(Login);
