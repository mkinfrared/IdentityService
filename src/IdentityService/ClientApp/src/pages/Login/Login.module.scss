.Login {
  background: url(../../images/mountain.jpg);
  background-size: cover;
  min-height: 100vh;
  width: 100%;

  .card {
    width: 300px;
  }

  .form {
    border-radius: 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-left: auto;
    min-height: 100vh;
  }
}
