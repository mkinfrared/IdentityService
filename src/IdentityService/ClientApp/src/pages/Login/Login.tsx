import { memo } from "react";

import LoginForm from "components/LoginForm";
import { useReturnUrl } from "hooks";

import css from "./Login.module.scss";

const Login = () => {
  const returnUrl = useReturnUrl();

  return (
    <div className={css.Login}>
      <LoginForm className={css.form} returnUrl={returnUrl} />
    </div>
  );
};

export { Login };

export default memo(Login);
