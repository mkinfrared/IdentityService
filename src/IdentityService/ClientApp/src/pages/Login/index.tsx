import ReactDOM from "react-dom";

import LoginFC from "./Login";

import "@identity-service/ui/lib/styles.min.css";
import "index.scss";

ReactDOM.render(<LoginFC />, document.getElementById("client-app"));
