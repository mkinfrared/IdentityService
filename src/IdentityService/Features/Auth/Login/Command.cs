using MediatR;

using Microsoft.AspNetCore.Identity;

namespace IdentityService.Features.Auth.Login
{
    public partial class Login
    {
        public class Command : IRequest<SignInResult>
        {
            public Command(string username, string password, string returnUrl)
            {
                Username = username;
                Password = password;
                ReturnUrl = returnUrl;
            }

            public string Username { get; }
            public string Password { get; }
            public string ReturnUrl { get; }
        }
    }
}
