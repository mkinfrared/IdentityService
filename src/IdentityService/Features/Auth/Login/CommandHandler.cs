using System.Threading;
using System.Threading.Tasks;

using IdentityService.Entities;

using MediatR;

using Microsoft.AspNetCore.Identity;

namespace IdentityService.Features.Auth.Login
{
    public partial class Login
    {
        public class CommandHandler : IRequestHandler<Command, SignInResult>
        {
            private readonly SignInManager<User> _signInManager;
            private readonly UserManager<User> _userManager;

            public CommandHandler(UserManager<User> userManager, SignInManager<User> signInManager)
            {
                _userManager = userManager;
                _signInManager = signInManager;
            }

            public async Task<SignInResult> Handle(Command request,
                CancellationToken cancellationToken)
            {
                var user = await _userManager.FindByNameAsync(request.Username);

                if (user == null)
                {
                    return SignInResult.Failed;
                }

                var signInResult = await
                    _signInManager.PasswordSignInAsync(user, request.Password, true, false);

                return signInResult;
            }
        }
    }
}
