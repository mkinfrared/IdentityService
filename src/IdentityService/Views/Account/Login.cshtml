@page
@model object

@{
    ViewBag.Title = "Login";
    ViewBag.Error = "Error message";
    Layout = "_Layout";
}

@section Head
{
    <link asp-href-include="~/login.*.css" rel="stylesheet">
}

@section PostBody
{
    @* src="~/login.js" *@
    <script type="application/javascript" asp-src-include="~/login.*.js"></script>
}

<div id="login-form">
    Loading...
</div>