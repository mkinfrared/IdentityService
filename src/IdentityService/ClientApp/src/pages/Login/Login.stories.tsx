import { Meta, Story } from "@storybook/react";

import { createMemoryHistory } from "utils/helpers";
import { withRouter } from "utils/storybookDecorators";

import { Login } from "./Login";

const history = createMemoryHistory();

history.push("/login");

export default {
  title: "Pages/Login",
  component: Login,
  decorators: [withRouter(history)]
} as Meta;

const Template: Story = (args) => <Login {...args} />;
const Default = Template.bind({});

Default.args = {};

export { Default };
