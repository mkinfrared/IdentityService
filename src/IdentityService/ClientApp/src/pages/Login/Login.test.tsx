import { render } from "@testing-library/react";

import { Login } from "./Login";

jest.mock("api/mutations");

describe("<Login />", () => {
  const Component = <Login />;

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("should be defined", () => {
    expect(Login).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
