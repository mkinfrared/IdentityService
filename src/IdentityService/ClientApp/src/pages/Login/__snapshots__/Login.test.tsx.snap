// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should match the snapshot 1`] = `
<div>
  <div
    class="Login"
  >
    <div
      class="Card-module_Card__2zjDp LoginForm form"
      data-testid="Card"
    >
      <form
        method="post"
      >
        <h3
          class="Heading-module_Heading__iIC9P heading"
          data-testid="Heading"
        >
          Login
        </h3>
        <label
          class="TextField-module_TextField__3-bgB input"
          data-testid="TextField"
        >
          <h5
            class="Heading-module_Heading__iIC9P TextField-module_label__SBMQi"
            data-testid="Heading"
          >
            Username
          </h5>
          <div
            class="TextField-module_input__3FXIK"
          >
            <input
              name="username"
              type="text"
              value=""
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_prefix__2eBJg"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_suffix__2X_LC"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
          </div>
          <p
            class="Text-module_Text__3EbTM TextField-module_error__3xz6q"
            data-testid="Text"
          />
        </label>
        <label
          class="TextField-module_TextField__3-bgB input TextField-module_hasSuffix__3ZA6e"
          data-testid="TextField"
        >
          <h5
            class="Heading-module_Heading__iIC9P TextField-module_label__SBMQi"
            data-testid="Heading"
          >
            Password
          </h5>
          <div
            class="TextField-module_input__3FXIK"
          >
            <input
              name="password"
              type="password"
              value=""
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_prefix__2eBJg"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_suffix__2X_LC"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
            </div>
          </div>
          <p
            class="Text-module_Text__3EbTM TextField-module_error__3xz6q"
            data-testid="Text"
          />
        </label>
        <input
          hidden=""
          name="returnUrl"
          readonly=""
          type="text"
          value=""
        />
        <div
          class="buttonWrapper"
        >
          <button
            class="Button-module_Button__2YI99 Button-module_opaque__1DMAu"
            data-testid="Button"
            disabled=""
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
